### PR TITLE
upx 4.2.4

### PR DESCRIPTION
**Pull request by Maddimax**
UPX is still useful on macOS since it can compress Linux and Windows binaries.

### CODE REVIEW

**Comment by fxcoudert** [comment]
It still produces invalid executables on macOS, which is its most common use case, so I don't think we're likely to re-enable it as-is. We could make it Linux-only.

**Comment by fxcoudert** [comment]
If think the issue is not that it has a restricted set of targets, it is that it's buggy: instead of refusing to compress macOS executables, it produces invalid executables.

**Comment by carlocab** [comment]
Re-enabling this would require, at minimum, fixing the `brew test` failure on macOS 14, or else we won't be able to build any new bottles for this once Monterey goes away soon.

Though doing only that probably only grants this a stay of execution for another year, so `brew test` should probably be fixed for all macOS versions.

**Comment by Maddimax** [comment]
I've changed it to update upx to 4.2.4 as that version does not create broken executables anymore. Probably the test will still fail as the application returns "1" as exit code. I'm not a ruby pro, can we change the test to check that the call of upx fails instead?

**Comment by carlocab** [comment]
> can we change the test to check that the call of upx fails instead?

That would make the test wrong, since running `ups -V` with the just-built `upx` does not fail:
```
❯ upx -V
upx 4.2.4
[snip]

❯ echo $?
0
```
so the test legitimately catches the fact that running `upx` on itself results in a broken binary.

**Comment by Maddimax** [comment]
> > can we change the test to check that the call of upx fails instead?
> 
> That would make the test wrong, since running `upx -V` with the just-built `upx` does not fail:

I meant the actual call of upx `system bin/"upx", "-1", "--force-execve", "./upx"`, can we invert the test to check that it fails?

The binary afterwards should still run successfully testing that upx did not in fact break it.

**Comment by carlocab** [comment]
I suggest changing the test to this instead:
```diff
diff --git a/Formula/u/upx.rb b/Formula/u/upx.rb
index 6ebb8bf90f3..0bb878ab5d6 100644
--- a/Formula/u/upx.rb
+++ b/Formula/u/upx.rb
@@ -23,11 +23,7 @@ class Upx < Formula
   end
 
   test do
-    cp bin/"upx", "."
-    chmod 0755, "./upx"
-
-    system bin/"upx", "-1", "--force-execve", "./upx"
-    system "./upx", "-V" # make sure the binary we compressed works
-    system bin/"upx", "-d", "./upx"
+    system bin/"upx", "-1", "-o", "./hello", test_fixtures("elf/hello")
+    assert_path_exists testpath/"hello"
   end
 end
```
This way we try to compress an ELF file and check that it was successful.

**Comment by Maddimax** [comment]
> I suggest changing the test to this instead:

Done

**Comment by carlocab** [comment]
I think you might've forgotten to commit the changes?

**Comment by Maddimax** [comment]
You are correct, I missed adding the file.

**Review by carlocab — APPROVED** [review]
Thanks!

For reference: `upx` no longer supports compressing macOS binaries. It will error out when you try, and suggests `--force-macos` (which is still broken).

Given this it seems reasonable to ship for the functionality they do support.

File: Formula/u/upx.rb [hunk 1]
```diff
@@ -29,8 +31,8 @@ def install
     cp bin/"upx", "."
     chmod 0755, "./upx"
```
Comment (carlocab) [inline Formula/u/upx.rb:33]: Looks like you forgot to remove these lines too. Mind opening a follow-up PR?

Comment (Maddimax) [reply Formula/u/upx.rb:33]: Done: #188154

**Pull request merged**